src/extension/background-script/db.ts

```diff
@@ -1,11 +1,11 @@
 import Dexie from "dexie";
 import "fake-indexeddb/auto";
 import browser from "webextension-polyfill";
-import type { Allowance, Payment, Blocklist } from "~/types";
+import type { Allowance, Transaction, Blocklist } from "~/types";
 
 class DB extends Dexie {
   allowances: Dexie.Table<Allowance, number>;
-  payments: Dexie.Table<Payment, number>;
+  payments: Dexie.Table<Transaction, number>;
```

Comment:
Why that? I don't think this is correct The data in the table does not match all the attributes of `Transaction`

@escapedcat ?

Reply:
Good point. I thought this has been done for a reason but should have asked. @im-adithya can you elaborate?

Reply:
I was getting a TS error as we have the `payments` type set to `Transaction[]` in types:

https://github.com/getAlby/lightning-browser-extension/blob/84a2d8fe8b77cf4a7170d65b09bec163eb2d6246/src/extension/background-script/actions/allowances/get.js#L22


https://github.com/getAlby/lightning-browser-extension/blob/84a2d8fe8b77cf4a7170d65b09bec163eb2d6246/src/types.ts#L325

Reply:
And we have this component saying transactions specifically and also uses Transaction type, so I thought changing the payments type to Payment[] would be breaking things:
https://github.com/getAlby/lightning-browser-extension/blob/84a2d8fe8b77cf4a7170d65b09bec163eb2d6246/src/app/screens/Home/index.tsx#L197

Reply:
Thanks!  
I think the payments in the db and the payments attached to an allowance should have the type `payment`.  
Currently only the transaction-table is using `transaction` I think.  
Following does that path I [created a new PR](https://github.com/im-adithya/lightning-browser-extension/pull/2) which to extend this PR. Currently there are a lot of changes, but once this one is rebased with our master it should only have one commit left.
I rebased this PR but I can't force push onto this one due to missing permissions. Not sure why. I believe maintainers usually can push into contributor-forks 🤷